tests/io/test_text.py

```diff
@@ -61,3 +62,24 @@ def test_parse_wpc_surface_bulletin():
                                                 [-102, 32], [-103, 31]])
 
     assert all(df.valid == datetime(2021, 6, 28, 18, 0, 0))
+
+
+def test_negative_lat_highres():
+    """Test decoding of high res coordinates with negative latitude."""
+    coord = '-4920344'
+    lon, lat = _decode_coords(coord)
+    assert lon == -034.4
```

Comment:
I'm actually surprised this didn't error (it would for `034`):
```suggestion
    assert lon == -34.4
```